File: CardGameAPI/CardGameAPI/Models/Player.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CardGame.Models
{
    public class Player
    {
        [Key]
        public Guid Id { get; set; }
        public string UserName { get; set; }
        public DateTimeOffset LastActivity { get; set; }
        public bool Admin { get; set; }
        public int Wins { get; set; } = 0;
    }

    // Used to store a player's information in a game
    public class GamePlayer
    {
        public Player Player { get; set; }
        public int Gold { get; set; } = 0;
        public int ReputationPoints { get; set; } = 0;
        public List<Card> Cards { get; set; } = new();
        public int Order { get; set; }
        public List<int> Dice { get; set; }
    }
}

File: CardGameAPI/CardGameAPI/CardGameUI/src/app/shared/models/player.ts
import {ICard} from './card';
import {Dice} from './dice';

export interface IUser
{
    id: string;
    userName: string;
    lastActivity: Date;
    admin: boolean;
    wins: number;
}

export interface IGamePlayer
{
    player: IUser;
    gold: number;
    reputationPoints: number;
    cards: ICard[];
    order: number;
    dice: Dice[];
    leader: boolean;
}
